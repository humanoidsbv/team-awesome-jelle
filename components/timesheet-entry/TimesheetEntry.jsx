import React from 'react';

import PropTypes from 'prop-types';

import './timesheet-entry.scss';

import { calculateDuration } from '../../services/convert-time/convert-time';

class TimesheetEntry extends React.Component {
  static propTypes = {
    clientName: PropTypes.string.isRequired,
    endTime: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    onDelete: PropTypes.func.isRequired,
    startTime: PropTypes.string.isRequired
  }

  toggleDeleting = () => {
    this.setState(({ isEntryDeleting }) => ({ isEntryDeleting: !isEntryDeleting }));
  }

  handleDelete = () => {
    if (window.confirm('Are you sure you want to delete this item?')) {
      this.props.onDelete(this.props.id);
    }
  }

  render() {
    const {
      clientName, endTime, startTime
    } = this.props;

    return (
      <div className="timesheet-entry">
        <p className="timesheet-entry__employer">
          {clientName}
        </p>

        <button
          className="timesheet-entry__delete-button"
          onClick={this.handleDelete}
          type="submit"
        >
          <svg
            className="timesheet-entry__delete-icon"
          />
            Delete
        </button>
        <div className="timesheet-entry__time">
          <p className="timesheet-entry__time-range">
            {`${new Date(startTime).toLocaleTimeString('nl-NL', { hour: 'numeric', minute: 'numeric' })} - ${new Date(endTime).toLocaleTimeString('nl-NL', { hour: 'numeric', minute: 'numeric' })}`}
          </p>
          <p className="timesheet-entry__cumulative-time">
            {calculateDuration(startTime, endTime)}
          </p>
        </div>
      </div>
    );
  }
}

export default TimesheetEntry;
