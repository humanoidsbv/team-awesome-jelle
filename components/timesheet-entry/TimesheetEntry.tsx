import React from 'react';

import './timesheet-entry.scss';

import { calculateDuration } from '../../services/convert-time/convert-time';

import { TimesheetEntryModel } from '../../ducks/time-entries';


interface TimesheetEntryProps {
  timesheet: TimesheetEntryModel;
  onDelete;
}

const TimesheetEntry = ({timesheet, onDelete} : TimesheetEntryProps ) => {
  const handleDelete = () => {
    if (window.confirm('Are you sure you want to delete this item?')) {
      onDelete(timesheet.id);
    }
  }

  return (
    <div className="timesheet-entry">
      <p className="timesheet-entry__employer">
        {timesheet.clientName}
      </p>

      <button
        className="timesheet-entry__delete-button"
        onClick={handleDelete}
        type="submit"
      >
        <svg
          className="timesheet-entry__delete-icon"
        />
          Delete
      </button>
      <div className="timesheet-entry__time">
        <p className="timesheet-entry__time-range">
          {`${new Date(timesheet.startTime).toLocaleTimeString('nl-NL', { hour: 'numeric', minute: 'numeric' })} - ${new Date(timesheet.endTime).toLocaleTimeString('nl-NL', { hour: 'numeric', minute: 'numeric' })}`}
        </p>
        <p className="timesheet-entry__cumulative-time">
          {calculateDuration(timesheet.startTime, timesheet.endTime)}
        </p>
      </div>
    </div>
  )
}

export default TimesheetEntry;
