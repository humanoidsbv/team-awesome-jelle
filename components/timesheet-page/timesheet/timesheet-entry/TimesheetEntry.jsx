import React from 'react';

import './timesheet-entry.scss';

const TimesheetEntry = () => (
  <div className="timesheet-entry-wrapper timesheet-entry-wrapper__good">
    <p className="timesheet-entry__employer">
      Port of Rotterdam
    </p>
    <p className="timesheet-entry__date">
      09:00 - 17:00
    </p>
  </div>
);

export default TimesheetEntry;
