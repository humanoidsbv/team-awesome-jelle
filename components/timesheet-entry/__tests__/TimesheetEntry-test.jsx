import React from 'react';
import { shallow } from 'enzyme';

import TimesheetEntry from '../TimesheetEntry';

const timesheetEntry = {
  employer: 'Port of Rotterdam',
  endTime: '2018-12-13T18:30:00.000Z',
  id: 1,
  startTime: '2018-12-13T18:00:00.000Z'
};

const handleDeleteTimesheetEntry = (timesheetEntryId) => {
  this.props.onDeleteTimesheetEntry(timesheetEntryId);
};

it('Renders Entry Component correctly', () => {
  const wrapper = shallow(
    <TimesheetEntry
      employer={timesheetEntry.employer}
      endTime={timesheetEntry.endTime}
      id={timesheetEntry.id}
      onDelete={handleDeleteTimesheetEntry}
      startTime={timesheetEntry.startTime}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
