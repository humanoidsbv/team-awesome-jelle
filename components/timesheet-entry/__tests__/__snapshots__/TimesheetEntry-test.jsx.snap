// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Entry Component correctly 1`] = `
<div
  className="timesheet-entry-wrapper"
>
  <p
    className="timesheet-entry__employer"
  >
    Port of Rotterdam
  </p>
  <button
    className="timesheet-entry__delete-button"
    onClick={[Function]}
    type="submit"
  >
    <svg
      className="timesheet-entry__delete-icon"
    />
    Delete
  </button>
  <p
    className="timesheet-entry__time"
  >
    7:00 PM
           -
           7:30 PM
  </p>
</div>
`;
